lib/content_location_filter.js

```diff
@@ -3,31 +3,27 @@
 const HyperSwitch = require('hyperswitch');
 const URI = HyperSwitch.URI;
 const mwUtil = require('./mwUtil');
-const path = require('path');
 const P = require('bluebird');
 
 function isAbsoluteRedirect(location) {
     return /^https?:/.test(location);
 }
 
-function resolveRelativeRedirect(hyper, req, res) {
-    return mwUtil.getSiteInfo(hyper, req)
-    .then((siteInfo) => {
-        const pathArr = req.uri.path.slice(2).map(encodeURIComponent);
-        pathArr.push('..');
-        pathArr.push(res.headers.location);
-        return `${siteInfo.baseUri}/${path.join.apply(path, pathArr)}`;
-    });
+function resolveRelativeRedirect(req, res) {
+    const pathArr = req.uri.path.slice(0, -1).map(encodeURIComponent);
+    pathArr.unshift('');
+    pathArr.push(res.headers.location);
+    return pathArr.join('/');
 }
 
 function resolvedRedirectResponse(hyper, req, res) {
     let getContentURI;
     if (isAbsoluteRedirect(res.headers.location)) {
-        getContentURI = P.resolve(res.headers.location);
+        getContentURI = res.headers.location;
     } else {
-        getContentURI = resolveRelativeRedirect(hyper, req, res);
+        getContentURI = resolveRelativeRedirect(req, res);
     }
-    return getContentURI.then((contentURI) => hyper.request({
+    return P.resolve(getContentURI).then((contentURI) => hyper.request({
```

Comment:
You don't need to do P.resolve anymore, hyper.request returns a promise, so you can  simplify this to
```
return hyper.request({
   method: req.method,
    uri: new URI(contentURI)
})...
```